pkg/controller/elasticsearch/pdb/reconcile_with_roles_test.go

```diff
@@ -977,6 +860,24 @@ func Test_allowedDisruptionsForRole(t *testing.T) {
 			},
 			want: 0,
 		},
+		{
+			name: "green health: 1 disruption allowed for data sts that is not HA",
+			args: args{
+				es:              esv1.Elasticsearch{Status: esv1.ElasticsearchStatus{Health: esv1.ElasticsearchGreenHealth}},
+				role:            []esv1.NodeRole{esv1.DataRole},
+				allStatefulSets: sset.StatefulSetList{ssetfixtures.TestSset{Replicas: 1}.Build()},
+			},
+			want: 1,
+		},
+		{
+			name: "green health: 1 disruption allowed for data sts that is HA",
+			args: args{
+				es:              esv1.Elasticsearch{Status: esv1.ElasticsearchStatus{Health: esv1.ElasticsearchGreenHealth}},
+				role:            []esv1.NodeRole{esv1.DataRole},
+				allStatefulSets: sset.StatefulSetList{ssetfixtures.TestSset{Replicas: 3}.Build()},
```

Comment:
For data nodes 2 can already be considered HA: one for the primary one for the replica